cpp/src/cylon/groupby/groupby_hash.hpp

```diff
@@ -0,0 +1,289 @@
+#include "groupby_aggregate_ops.hpp"
+
+namespace cylon {
+
+template<typename T, cylon::GroupByAggregationOp AggregateOp,
+    typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+struct AggregateKernel {
+
+};
+
+template<typename T>
+struct AggregateKernel<T, cylon::GroupByAggregationOp::SUM> {
+  using HashMapType = std::tuple<T>;
+  using ResultType = T;
+  using ResultArrowType = typename arrow::CTypeTraits<T>::ArrowType;
+
+//  static constexpr const char* _prefix = "sum_";
+
+  static constexpr HashMapType Init(const T &value) {
+    return HashMapType{value};
+  }
+
+  static inline void Update(const T &value, HashMapType *result) {
+    std::get<0>(*result) += value;
+  }
+
+  static inline ResultType Finalize(const HashMapType *result) {
+    return std::get<0>(*result);
+  }
+};
+
+template<typename T>
+struct AggregateKernel<T, cylon::GroupByAggregationOp::MIN> {
+  using HashMapType = std::tuple<T>;
+  using ResultType = T;
+  using ResultArrowType = typename arrow::CTypeTraits<T>::ArrowType;
+
+//  static constexpr const char* _prefix = "min_";
+
+  static constexpr HashMapType Init(const T &value) {
+    return HashMapType{value};
+  }
+
+  static inline void Update(const T &value, HashMapType *result) {
+    std::get<0>(*result) = std::min(value, std::get<0>(*result));
+  }
+
+  static inline ResultType Finalize(const HashMapType *result) {
+    return std::get<0>(*result);
+  }
+};
+
+template<typename T>
+struct AggregateKernel<T, cylon::GroupByAggregationOp::MAX> {
+  using HashMapType = std::tuple<T>;
+  using ResultType = T;
+  using ResultArrowType = typename arrow::CTypeTraits<T>::ArrowType;
+
+//  static constexpr const char* _prefix = "max_";
+
+  static constexpr HashMapType Init(const T &value) {
+    return HashMapType{value};
+  }
+
+  static inline void Update(const T &value, HashMapType *result) {
+    std::get<0>(*result) = std::max(value, std::get<0>(*result));
+  }
+
+  static inline ResultType Finalize(const HashMapType *result) {
+    return std::get<0>(*result);
+  }
+};
+
+template<typename T>
+struct AggregateKernel<T, GroupByAggregationOp::COUNT> {
+  using HashMapType = std::tuple<int64_t>;
+  using ResultType = int64_t;
+  using ResultArrowType = arrow::Int64Type;
+
+//  static constexpr const char* _prefix = "count_";
+
+  static constexpr HashMapType Init(const T &value) {
+    return HashMapType{1};
+  }
+
+  static inline void Update(const T &value, HashMapType *result) {
+    std::get<0>(*result) += 1;
+  }
+
+  static inline ResultType Finalize(const HashMapType *result) {
+    return std::get<0>(*result);
+  }
+};
+
+//template<typename T>
+//struct AggregateKernel<T, GroupByAggregationOp::MEAN> {
+//  using HashMapType = std::tuple<T, int64_t>;
+//  using ResultType = T;
+//  using ResultArrowType = typename arrow::CTypeTraits<T>::ArrowType;
+//
+////  static constexpr char _prefix[] = "mean_";
+//
+//  static constexpr HashMapType Init(const T &value) {
```

Comment:
Remove unwanted code

Reply:
I am going to keep this part. This is the mean kernel. Didn't add it yet because I still don't have a proper API for multi value merge.